controllers/vspherevm_controller.go

```diff
@@ -138,6 +139,43 @@ func (r vmReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error)
 		return reconcile.Result{}, err
 	}
 
+	// Fetch the owner VSphereMachine.
+	vsphereMachine := &infrav1.VSphereMachine{}
+	hasMachineOwner := clusterutilv1.HasOwnerRef(vsphereVM.OwnerReferences, metav1.OwnerReference{
```

Comment:
This breaks `HAProxyLoadBalancer` in v1alpha3 setups. the haproxy controller generates a `VSphereVM`, so a `VSphereVM` can have another ownerRef than a `VSphereMachine`